typing/typecore.ml

```diff
@@ -850,13 +850,17 @@ and build_as_type_aux (env : Env.t) p =
       newty (Ttuple labeled_tyl)
   | Tpat_construct(_, cstr, pl, vto) ->
       let keep =
-        cstr.cstr_private = Private || cstr.cstr_existentials <> [] ||
+        cstr.cstr_private = Private ||
         vto <> None (* be lazy and keep the type for node constraints *) in
       if keep then p.pat_type else
       let tyl = List.map (build_as_type env) pl in
       let ty_args, ty_res, _ =
-        instance_constructor Keep_existentials_flexible cstr
-      in
+        instance_constructor Keep_existentials_flexible cstr in
+      (* [p] is a valid result of type inference, and [ty_args] is a generic
+         instance where existentials are variables, with non-escaping ones
+         are at generic level.
```

Comment:
I don't understand "with non-escaping ones are at generic level" -- there is a small grammar issue but I am also not sure what you mean.

Reply:
Updated the comment, as it did not match the code anymore.